Client/game_sa/gamesa_init.h

```diff
@@ -104,3 +104,42 @@ void MemOrFast(U ptr, const T value)
 
 bool GetDebugIdEnabled(uint uiDebugId);
 void LogEvent(uint uiDebugId, const char* szType, const char* szContext, const char* szBody, uint uiAddReportLogId = 0);
+
+namespace CallingConventions
+{
+    struct thiscall{};
+    struct ___cdecl{};
+    struct stdcall{};
+    struct fastcall{};
+    struct vectorcall{};
+} // namespace CallingConventions
+
+template <typename ReturnType, typename CallingConvention, typename Func, typename... Args>
+typename std::enable_if<std::is_same<CallingConvention, CallingConventions::thiscall>::value, ReturnType>::type CallGTAFunction(Func function, Args... params)
```

Comment:
Please, use universal references and the perfect forwarding.

```
template <typename ReturnType, typename CallingConvention, typename Func, typename... Args>
typename std::enable_if<std::is_same<CallingConvention, CallingConventions::thiscall>::value, ReturnType>::type CallGTAFunction(Func function, Args&&... params)
{
    return reinterpret_cast<ReturnType(__thiscall*)(Args...)>(function)(std::forward<Args>(params)...);
}
```
The same for below.

Reply:
Universal references are probably not a good idea? Then the argument, which is a pointer, turns into a reference
``FxSystem_c* -> FxSystem_c*&`` and this causes crashes

Reply:
Why would it turn into a reference? universal refs + perfect forwarding are created to perfectly mirror types.